rest-service/manager_rest/rest/responses.py

```diff
@@ -152,18 +154,30 @@ def __init__(self, **kwargs):
         self.description = kwargs.get('description')
 
 
-@swagger.model
-class Label(object):
+# This has to be a constant because it's needed by the resource_models, and
+# won't exist on the class until instantiated.
+LABEL_FIELDS = {'key': fields.String,
+                'value': fields.String,
+                'created_at': fields.String,
+                'created_by': fields.String}
 
-    resource_fields = {
-        'key': fields.String,
-        'value': fields.String,
-        'created_at': fields.String,
-        'created_by': fields.String,
-    }
 
-    def __init__(self, **kwargs):
-        self.key = kwargs.get('key')
-        self.value = kwargs.get('value')
-        self.created_at = kwargs.get('created_at')
-        self.created_by = kwargs.get('created_by')
+@swagger.model
+@dataclass(kw_only=True, unsafe_hash=True)
+class Label:
+    """A blueprint or deployment label."""
+    key: str
+    value: str
+    created_at: Optional[str] = field(default=None, compare=False)
+    created_by: Optional[str] = field(default=None, compare=False)
+
+    resource_fields: dict = field(
+        default_factory=lambda: LABEL_FIELDS,
+        compare=False, repr=False, init=False,
+    )
```

Comment:
this is a bad idea and let's not do this - adds some runtime overhead for no good reason.

Instead, class attributes are to be type-declared as `ClassVar` (or we can be even more specific here with a `ClassVar[Dict[str, Any]]` - but the values are indeed going to need to be Any...)

Reply:
Does that withold them from the compare? Not that it matters, I guess... but it will still put them in the repr, which sucks.

Reply:
Then again, I'm not sure we actually need resource_fields to be on the Label at all, checking at the mo.

Reply:
OK, it is ClassVar[Mapping]ed up.